core/core_bind.cpp

```diff
@@ -137,9 +137,13 @@ void ResourceLoader::_bind_methods() {
 
 ////// ResourceSaver //////
 
-Error ResourceSaver::save(const String &p_path, const Ref<Resource> &p_resource, uint32_t p_flags) {
-	ERR_FAIL_COND_V_MSG(p_resource.is_null(), ERR_INVALID_PARAMETER, "Can't save empty resource to path '" + String(p_path) + "'.");
-	return ::ResourceSaver::save(p_path, p_resource, p_flags);
+Error ResourceSaver::save(const Ref<Resource> &p_resource, const String &p_path, uint32_t p_flags) {
+	String path = p_path;
+	ERR_FAIL_COND_V_MSG(p_resource.is_null(), ERR_INVALID_PARAMETER, "Can't save empty resource to path '" + path + "'.");
+	if (path.is_empty()) {
+		path = p_resource->get_path();
```

Comment:
What if resource's own path is empty? I guess empty path wasn't handled here before neither... Is that handled down the line?

Reply:
Resource path can only be empty when it's a freshly created built-in resource until the scene is saved.

I checked how it was handled before and the method just fails silently.

Reply:
I mean, any freshly created resource that only exists in memory wouldn't have a path. Failing silently is not optimal, though. Maybe worth checking?

I'm not sure if it's left this way in case a specific saver implements automatic name generation, or something...

Reply:
I added an error when path is empty. There doesn't seem to be any check for empty paths in the actual saver and seems like it just happily saves the file under empty path, which is successful, but has no effect, because the file does not exist. Kind of reminds me of #41623 (file operations just allow stuff like this)